geomesa-hbase/geomesa-hbase-rpc/src/main/scala/org/locationtech/geomesa/hbase/rpc/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -134,31 +147,48 @@ object GeoMesaCoprocessor extends LazyLogging {
 
     private val coprocessor = CachedThreadPool.submit(new Runnable() {
       override def run(): Unit = {
+        val callback = new GeoMesaHBaseCallBack(resultQueue)
         try {
           if (!closed.get) {
+            logger.trace(s"Calling htable.coproService (first time): " +
+              s"${ByteArrays.printable(scan.getStartRow)} to ${ByteArrays.printable(scan.getStopRow)}")
             htable.coprocessorService(service, scan.getStartRow, scan.getStopRow, callable, callback)
           }
+          logger.trace(s"Closed: ${closed.get}. Callback.lastRow: ${ByteArrays.printable(callback.lastRow)}")
+
+          // If the scan hasn't been killed and we are not done, then re-issue the request.
+          while (!closed.get() && callback.lastRow != null) {
+            // Reset the callback's status
+            callback.lastRow = null
+
+            val nextRow = ByteArrays.rowFollowingRow(callback.lastRow)
+            logger.trace(s"Scan continuing from next row is ${ByteArrays.printable(nextRow)}")
+
+            scan.setStartRow(nextRow)
```

Comment:
maybe we can avoid mutating the scan, and also having to regenerate the request - since we're calling `htable.coprocessorService` with the new start row, it shouldn't matter, right?

Reply:
if that works, we could make `callable` back into a `val` too

Reply:
Ok, doing that did seem to work.  I did have that change in place when I tried the while loop.

Reply:
we can remove this line then right? `scan.setStartRow(nextRow)`